src/foam/dao/index/TreeIndex.java

```diff
@@ -136,7 +137,9 @@ public SelectPlan planSelect(Object state, Sink sink, long skip, long limit, Com
     Object[] statePredicate = simplifyPredicate(state, predicate);
     state = statePredicate[0];
     predicate = (Predicate) statePredicate[1];
-    if ( predicate == null && sink instanceof GroupBy )
+    if ( predicate == null && sink instanceof GroupBy
+        && ( (GroupBy) sink ).getArg1().toString().equals(prop_.toString())
+        && order == null && skip == 0 && limit == AbstractDAO.MAX_SAFE_INTEGER )
       return new GroupByPlan(state, sink, skip, limit, order, predicate, prop_, tail_);
```

Comment:
Don't pass skip and limit to GroupByPlan since it doesn't use it.

Reply:
Ok, I will fix that.